core/blockchain.go

```diff
@@ -1053,9 +1055,47 @@ func (bc *BlockChain) Accept(block *types.Block) error {
 	bc.addAcceptorQueue(block)
 	acceptedBlockGasUsedCounter.Inc(int64(block.GasUsed()))
 	acceptedTxsCounter.Inc(int64(len(block.Transactions())))
+	if baseFee := block.BaseFee(); baseFee != nil {
+		blockGasPriceGauge.Update(baseFee.Int64())
+	}
+	total, err := TotalFees(block, bc.GetReceiptsByHash(block.Hash()))
+	if err != nil {
+		log.Debug(fmt.Sprintf("TotalFees %s", err))
+	} else {
+		blockTotalFeesGauge.Update(total.Int64())
+	}
 	return nil
 }
 
+// TotalFees computes total consumed fees in ETH. Block transactions and receipts have to have the same order.
+func TotalFees(block *types.Block, receipts []*types.Receipt) (*big.Int, error) {
+	baseFee := block.BaseFee()
+	feesWei := new(big.Int)
+	if len(block.Transactions()) != len(receipts) {
+		return nil, errors.New("mismatch between total number of transactions and receipts")
+	}
+	for i, tx := range block.Transactions() {
+		var minerFee *big.Int
+		if baseFee == nil {
+			// legacy block, no baseFee
+			minerFee = tx.GasPrice()
+		} else {
+			minerFee = new(big.Int).Add(baseFee, tx.EffectiveGasTipValue(baseFee))
+		}
+		feesWei.Add(feesWei, new(big.Int).Mul(new(big.Int).SetUint64(receipts[i].GasUsed), minerFee))
+	}
+	return feesWei, nil
+}
+
+func TotalFeesFloat(block *types.Block, receipts []*types.Receipt) *big.Float {
```

Comment:
Could we change the return type here to return the error instead of a debug log and zero?